external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -147,6 +147,21 @@ private[sql] class AvroDeserializer(
           s"Avro logical type $other cannot be converted to SQL type ${TimestampType.sql}.")
       }
 
+      case (LONG, TimestampNTZType) => avroType.getLogicalType match {
+        // For backward compatibility, if the Avro type is Long and it is not logical type
```

Comment:
do we have backport compatibility issues here? I think we should fail when we read AVRO long as TS NTZ type.

Reply:
I think @beliefer is trying to make it consistent with the conversion between `TimestampType` and `LongType`. 
For older Avro version, there are no logical types. So previously I make it convertible for `TimestampType` and `LongType`.
I don't have a good idea about this either.

Reply:
Yes. As @gengliangwang said, I think we should follows the way treat `null` as timestamp without time zone type with millisecond precision.

Reply:
then can we at least update the comment? it's no thing about backward compatibility, but to keep consistent with TS LTZ.

Reply:
OK.